apis/apigee/v1alpha1/environmentgroup_types.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refv1alpha1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1alpha1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var ApigeeEnvgroupGVK = GroupVersion.WithKind("ApigeeEnvgroup")
+
+type Parent struct {
+	// +required
+	OrganizationRef *refv1alpha1.OrganizationRef `json:"organizationRef"`
+}
+
+// ApigeeEnvgroupSpec defines the desired state of ApigeeEnvgroup
+// +kcc:proto=mockgcp.cloud.apigee.v1.GoogleCloudApigeeV1EnvironmentGroup
+type ApigeeEnvgroupSpec struct {
+	Parent `json:",inline"`
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceID field is immutable"
```

Comment:
This needs to be removed, template is getting updated in https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/3434. Also remove the `// Immutable.` comment. We are allowing changes to this field, so the user can set from empty string -> the actual resource ID, if they want to. Otherwise, it is immutable.

You will need to run `make manifests` (and maybe `make resource-docs`) after this change also.

Reply:
Done